src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -0,0 +1,741 @@
+        if (!impStackTop(0).val->IsIntegralConst(4)) // StringComparison.Ordinal
+        {
+            // TODO-Unroll-CQ: Unroll & vectorize OrdinalIgnoreCase
+            return nullptr;
+        }
+        op1 = impStackTop(2).val;
+        op2 = impStackTop(1).val;
+    }
+    else
+    {
+        assert(argsCount == 2);
+        op1 = impStackTop(1).val;
+        op2 = impStackTop(0).val;
+    }
+
+    // For generic StartsWith and Equals we need to make sure T is char
+    if (sig->sigInst.methInstCount != 0)
+    {
+        assert(sig->sigInst.methInstCount == 1);
+        CORINFO_CLASS_HANDLE targetElemHnd = sig->sigInst.methInst[0];
+        CorInfoType          typ           = info.compCompHnd->getTypeForPrimitiveValueClass(targetElemHnd);
+        if ((typ != CORINFO_TYPE_SHORT) && (typ != CORINFO_TYPE_USHORT) && (typ != CORINFO_TYPE_CHAR))
+        {
+            return nullptr;
+        }
+    }
+
+    // Try to obtain original string literals out of span arguments
+    GenTreeStrCon* op1Str = impGetStrConFromSpan(op1);
+    GenTreeStrCon* op2Str = impGetStrConFromSpan(op2);
+
+    if (!((op1Str != nullptr) ^ (op2Str != nullptr)))
+    {
+        // either op1 or op2 has to be '(ReadOnlySpan)"cns"'
+        return nullptr;
+    }
+
+    GenTree*       spanObj;
+    GenTreeStrCon* cnsStr;
+    if (op1Str != nullptr)
+    {
+        cnsStr  = op1Str;
+        spanObj = op2;
+    }
+    else
+    {
+        cnsStr  = op2Str;
+        spanObj = op1;
+    }
+
+    int             cnsLength = -1;
+    const char16_t* str       = nullptr;
+    if (cnsStr->IsStringEmptyField())
+    {
+        // check for fake "" first
+        cnsLength = 0;
+        JITDUMP("Trying to unroll MemoryExtensions.Equals|SequenceEqual|StartsWith(op1, \"\")...\n", str)
+    }
+    else
+    {
+        str = info.compCompHnd->getStringLiteral(cnsStr->gtScpHnd, cnsStr->gtSconCPX, &cnsLength);
+        if (cnsLength < 0 || str == nullptr)
+        {
+            // We were unable to get the literal (e.g. dynamic context)
+            return nullptr;
+        }
+        JITDUMP("Trying to unroll MemoryExtensions.Equals|SequenceEqual|StartsWith(op1, \"%ws\")...\n", str)
+    }
+
+    CORINFO_CLASS_HANDLE spanCls      = gtGetStructHandle(spanObj);
+    CORINFO_FIELD_HANDLE pointerHnd   = info.compCompHnd->getFieldInClass(spanCls, 0);
+    CORINFO_FIELD_HANDLE lengthHnd    = info.compCompHnd->getFieldInClass(spanCls, 1);
+    const unsigned       lengthOffset = info.compCompHnd->getFieldOffset(lengthHnd);
+
+    // Create a placeholder for Span object - we're not going to Append it to statements
+    // in advance to avoid redundant spills in case if we fail to vectorize
+    unsigned spanObjRef         = lvaGrabTemp(true DEBUGARG("spanObjRef"));
+    lvaTable[spanObjRef].lvType = TYP_BYREF;
+
+    GenTree* spanObjRefLcl = gtNewLclvNode(spanObjRef, TYP_BYREF);
+    GenTree* spanLength    = gtNewFieldRef(TYP_INT, lengthHnd, gtClone(spanObjRefLcl), lengthOffset);
```

Comment:
Or if you only see both fields and locals then at least verify in the above that `gtClone` doesn't fail.